packages/plugin-interactive-tools/sources/commands/upgrade-interactive.tsx

```diff
@@ -172,46 +176,58 @@ export default class UpgradeInteractiveCommand extends BaseCommand {
             </Text>
           </Box>
           <Box width={17}><Text bold underline color="gray">Current</Text></Box>
-          <Box width={17}><Text bold underline color="gray">Range/Latest</Text></Box>
+          <Box width={17}><Text bold underline color="gray">Range</Text></Box>
+          <Box width={17}><Text bold underline color="gray">Latest</Text></Box>
         </Box>
       );
     };
 
-    const UpgradeEntry = ({active, descriptor}: {active: boolean, descriptor: Descriptor}) => {
+    const UpgradeEntry = ({active, descriptor, suggestions}: {active: boolean, descriptor: Descriptor, suggestions: Array<UpgradeSuggestion>}) => {
       const [action, setAction] = useMinistore<string | null>(descriptor.descriptorHash, null);
-      const [suggestions, setSuggestions] = useState<Array<{value: string | null, label: string}> | null>(null);
 
-      const mountedRef = useRef<boolean>(true);
+      const packageIdentifier = structUtils.stringifyIdent(descriptor);
+      const padLength = Math.max(0, 45 - packageIdentifier.length);
+      return <>
+        <Box>
+          <Box width={45}>
+            <Text bold>
+              {structUtils.prettyIdent(configuration, descriptor)}
+            </Text>
+            <Pad active={active} length={padLength}/>
+          </Box>
+          {suggestions !== null
+            ? <ItemOptions active={active} options={suggestions} value={action} skewer={true} onChange={setAction} sizes={[17, 17, 17]} />
+            : <Box marginLeft={2}><Text color="gray">Fetching suggestions...</Text></Box>
+          }
+        </Box>
+      </>;
+    };
+
+    const UpgradeEntries = ({dependencies}: { dependencies: Array<Descriptor> }) => {
+      const [suggestions, setSuggestions] = useState<Array<readonly [Descriptor, UpgradeSuggestions]>|null>(null);
 
       useEffect(() => {
-        return () => {
-          mountedRef.current = false;
-        };
+        Promise.all(dependencies.map(descriptor => fetchSuggestions(descriptor)))
+          .then(allSuggestions => {
+            const mappedToSuggestions = dependencies.map((descriptor, i) => {
+              const suggestionsForDescriptor = allSuggestions[i];
+              return [descriptor, suggestionsForDescriptor] as const;
+            }).filter(([_, suggestions]) => suggestions.length > 1);
+
+            setSuggestions(mappedToSuggestions);
```

Comment:
You shouldn't have removed the `mountedRef.current` check; this will now be susceptible to the type of warning React shouts when calling `setState` on unmounted components :(

(I know, it's unlikely to happen since the suggestions will probably be finished before the component is unmounted, and even if not the unmount probably mean the process will be shutdown, but I don't like having known bugs, however unlikely :p)

Reply:
Ah, k. My thinking there was that since it was loading all of the information up-front we wouldn't have the issue with the data coming back after a row had scrolled off screen.

Reply:
Updated.

Reply:
Perfect, thanks!

Reply:
You're welcome! Thanks for accepting my changes. :D